src/Config/ConfigFile.php

```diff
@@ -0,0 +1,182 @@
+<?php namespace Winter\Storm\Config;
+
+use PhpParser\Error;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Stmt;
+use PhpParser\ParserFactory;
+use PhpParser\PrettyPrinterAbstract;
+
+/**
+ * Class ConfigFile
+ * @package Winter\Storm\Config
+ */
+class ConfigFile
+{
+    /**
+     * @var null
+     */
+    protected $ast = null;
+    /**
+     * @var string|null
+     */
+    protected $file = null;
+    /**
+     * @var PrettyPrinterAbstract|WinterPrinter|null
+     */
+    protected $printer = null;
+
+    /**
+     * Config constructor.
+     * @param $ast
+     * @param PrettyPrinterAbstract|null $printer
+     */
+    public function __construct($ast, string $file = null, PrettyPrinterAbstract $printer = null)
```

Comment:
@jaxwilko given the check immediately below, would the `$ast` be an array?
```suggestion
    public function __construct(array $ast, string $file = null, PrettyPrinterAbstract $printer = null)
```

Reply:
I'll have to check later but I'm pretty sure the parser returns mixed results, Ideally we want to take whatever we're given and then validate it inside the constructor to ensure it's a config style php file

Reply:
I double checked and array should be fine, I've added this in c09718739344ec6544dc690fa48d11de85b52a78